dev/AnimatedVisualPlayer/AnimatedVisualPlayer.cpp

```diff
@@ -342,10 +343,14 @@ AnimatedVisualPlayer::AnimatedVisualPlayer()
 AnimatedVisualPlayer::~AnimatedVisualPlayer()
 {
     // Ensure any outstanding play is stopped.
-    // NOTE: Stop() can cause reentrance when clients react to DP changes, but because
-    //       we're in the destructor we know that there aren't any clients who can reach
-    //       us, so reentrance is not a concern. 
-    Stop();
+    if (m_nowPlaying)
+    {
+        // To stop and destroy m_nowPlaying we need to call Complete()
+        // We need to detach m_nowPlaying first so that Complete()
+        // will not try to call us (owner), since we already in the destructor
+        auto nowPlaying = std::move(m_nowPlaying);
```

Comment:
How does the move "detach" it? m_nowPlaying still points back at AnimatedVisualPlayer, right?

Reply:
std::move leaves std::shared_ptr in empty state (nullptr)

Reply:
After the move, m_nowPlaying is null
Thus when the helper checks IsCurrentPlay(), that will return false and avoid calling back into the AVP to update its state.

Reply:
But it still calls m_owner to do so. So if NowPlaying outlives AnimatedVisualPlayer, we're still in trouble.

Reply:
`nowPlaying->Complete()` is called inside this destructor, so [this check](https://github.com/microsoft/microsoft-ui-xaml/pull/4955/commits/a2dadc322c90ea5a7b46bbc923357f051fed5ba8#R144) will happen before destructor actually destroys the object?

Reply:
Yes but it's an accident waiting to happen. It might be that the code today never touches m_owner after this point but what if it does? Can we instead make the code clear the m_owner field out of m_nowPlaying?

```
m_nowPlaying->m_owner = nullptr;
// Call complete after clearing m_owner so we don't get called back to update IsPlaying
m_nowPlaying->Complete();
```

Then when this object finishes destructing it'll release its shared_ptr ref on m_nowPlaying and it's obvious from the code that we are preventing a dangling pointer situtaion.

Reply:
I see..., yeah it makes sense